projects/angular-components/src/lib/datatable/datatable.component.ts

```diff
@@ -0,0 +1,297 @@
+import { AfterViewInit, Component, ContentChild, Input, OnDestroy } from '@angular/core';
+import { CdkTableModule, DataSource } from '@angular/cdk/table';
+import { CommonModule } from '@angular/common';
+import { FormsModule } from '@angular/forms';
+import { BehaviorSubject, Observable, Subscription } from 'rxjs';
+import { DtContentDirective } from './dt-content.directive';
+
+export type TableOptions = {
+  sizeOptions: number[];
+  size: number;
+  serverSide: boolean;
+  filter: boolean;
+};
+
+export type DataTableColumn<T> = {
+  name: string;
+  displayName: string;
+  property: keyof T | null;
+
+  html?: boolean;
+  className?: string;
+  hidden?: boolean;
+};
+
+export type DataTableEntryInfo = {
+  minPageEntry: number;
+  maxPageEntry: number;
+  totalFilteredEntries: number;
+  totalEntries: number;
+};
+
+export type DataTablePaginationInfo = {
+  currentPage: number;
+  totalPages: number;
+};
+
+export type DataTableServerRequestInfo = {
+  size: number;
+  offset: number;
+  sort: 'asc' | 'desc';
+};
+
+export type DataTableServerResponseInfo<T> = {
+  totalEntries: number;
+  filteredEntries: number;
+  size: number;
+  offset: number;
+  data: T[];
+};
+
+@Component({
+  selector: 'ff-datatable',
+  standalone: true,
+  imports: [CommonModule, FormsModule, CdkTableModule],
+  templateUrl: './datatable.component.html',
+  styleUrl: './datatable.component.scss',
+})
+export class DatatableComponent<T> implements AfterViewInit, OnDestroy {
+  @Input({ required: true }) public datasource!: DataTableDataSource<T>;
+  @Input({ required: true }) public displayColumns: DataTableColumn<T>[] = [];
+
+  @ContentChild(DtContentDirective) protected content!: DtContentDirective<T>;
+  protected totalFilteredEntries: number = 0;
+  protected totalEntries: number = 0;
+  protected minPageEntry: number = 0;
+  protected maxPageEntry: number = 0;
+  protected currentPage: number = 1;
+  protected totalPages: number = 0;
+
+  protected get displayedColumns(): string[] {
+    return this.displayColumns.map((column) => column.name);
+  }
```

Comment:
I have never thought about doing it this way, I'd say its a little less explicit but since it's already obvious what the map item is I'd say it looks good